src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/repository/UserPersonDetailRepositoryTest.kt

```diff
@@ -0,0 +1,194 @@
+package uk.gov.justice.digital.hmpps.nomisuserrolesapi.jpa.repository
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.test.context.ActiveProfiles
+import uk.gov.justice.digital.hmpps.nomisuserrolesapi.helper.generalUserEntityCreator
+import uk.gov.justice.digital.hmpps.nomisuserrolesapi.helper.localAdministratorEntityCreator
+import uk.gov.justice.digital.hmpps.nomisuserrolesapi.jpa.LAAAdminUser
+import uk.gov.justice.digital.hmpps.nomisuserrolesapi.jpa.LAAGeneralUser
+import java.time.LocalDate
+import javax.persistence.EntityManager
+
+@DataJpaTest
+@ActiveProfiles("test")
+class UserPersonDetailRepositoryTest {
+  @Autowired
+  lateinit var repository: UserPersonDetailRepository
+
+  @Autowired
+  lateinit var localAdminAuthorityRepository: LocalAdminAuthorityRepository
+
+  @Autowired
+  lateinit var caseloadRepository: CaseloadRepository
+
+  @Autowired
+  lateinit var entityManager: EntityManager
+
+  @Test
+  internal fun `can read a user general user`() {
+    assertThat(repository.findByIdOrNull("jim.bubbles")).isNull()
+
+    generalUserEntityCreator(repository, localAdminAuthorityRepository, caseloadRepository)
+      .username("jim.bubbles")
+      .atPrison("WWI")
+      .buildAndSave()
+
+    entityManager.clear()
```

Comment:
required to force the test to go back to database for reads rather than the hibernation session that would have cached the entity